python_modules/dagster/dagster/core/definitions/decorators/repository.py

```diff
@@ -23,6 +23,8 @@ def __init__(self, name: Optional[str] = None, description: Optional[str] = None
         self.description = check.opt_str_param(description, "description")
 
     def __call__(self, fn: Callable[[], Any]) -> RepositoryDefinition:
+        from dagster.core.asset_defs import ForeignAsset
```

Comment:
we have to include foreign assets on the repository directly because they don't belong to a job?  This seems very awkward... 

Is this a step towards the job-less assets on the repository?

Did you consider having `build_asset_job` take in a set of foreign assets instead?  I suppose we would then need to subclass it to be a more special `AssetJobDefinition` that can keep track of them.

Reply:
oh wait, we already do this...

Why do we need foreign assets on the repo?

Reply:
It's definitely a departure from the kinds of definitions we currently have on repositories.

The advantage of having these foreign assets as standalone objects in repositories instead of inside jobs is that it would enable users to define them without defining jobs that consume them.

Example scenario: there's an analytics team that uses dbt + Airflow to create a bunch of tables in a database. There's also an ML team that wants to use Dagster to build ML models etc. on top of those tables.  The ML team writes a dbt utility that consumes the dbt project and creates a foreign asset for each model inside of it.  Now metadata for all those tables is available in Dagit, means that the ML users can find out info about all the assets they might care about in one place.

Reply:
ah, okay that makes sense.  Thanks for explaining.